supabase/functions/chat/index.ts

```diff
@@ -36,13 +65,69 @@ async function generateResponse(
   return choices[0].message;
 }
 
-const chat = async (req) => {
+async function prepareChatMessagesWithEmbeddingRecords(
```

Comment:
Logically I would probably make this function return `relevantRecords` and then construct `messages` outside of it. That way, the function only do one "logic"/"business" functionality

Reply:
whose comment are you talking about?